Shared.Tests/ApiTest.cs

```diff
@@ -1490,5 +1491,53 @@ public void TestSignParameters()
 
             StringAssert.AreNotEqualIgnoringCase(m_api.SignParameters(paramsSetOne), m_api.SignParameters(paramsSetTwo), "The signatures are equal.");
         }
+
+        [Test]
+        public void TestVerifyApiResponseSignature()
+        {
+            const string publicId = "resource id";
+            const string version = "resource version";
+            var responseParameters = new SortedDictionary<string, object>() {
+                { "public_id", publicId},
+                { "version", version},
+            };
+            var correctSignature = m_api.SignParameters(responseParameters);
+
+            Assert.IsTrue(m_api.VerifyApiResponseSignature(publicId, version, correctSignature), "The response signature is valid for the same parameters");
+
+            responseParameters["version"] = "new version";
+            var newVersionSignature = m_api.SignParameters(responseParameters);
+
+            Assert.IsFalse(m_api.VerifyApiResponseSignature(publicId, version, newVersionSignature), "The response signature is invalid for the wrong version");
+
+            responseParameters["version"] = version;
+            responseParameters["public_id"] = "another id";
+            var anotherResourceSignature = m_api.SignParameters(responseParameters);
+
+            Assert.IsFalse(m_api.VerifyApiResponseSignature(publicId, version, anotherResourceSignature), "The response signature is invalid for the wrong resource");
+        }
+
+        [Test]
+        public void TestVerifyNotificationSignature()
+        {
+            const string body = "some response body";
```

Comment:
We prefer test values to be as similar as possible to the real world values.( in this case some JSON response)